### PR TITLE
Add Pulse APIs

### PR DESCRIPTION
**Pull request by RahulBenaka**
### What does this Pull Request accomplish?

Adds Pulse support

### Why should this Pull Request be merged?
Adding this as a part of WI: [Generate  gRPC device code for RFmx Pulse](https://dev.azure.com/ni/DevCentral/_workitems/edit/2787959)

### What testing has been done?

PR + built output in grpc-device

### CODE REVIEW

File: source/codegen/metadata/nirfmxpulse/config.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
         "NIComplexSingle": "nidevice_grpc.NIComplexNumberF32",
         "NIComplexDouble": "nidevice_grpc.NIComplexNumber",
     },
-    'code_readiness': 'NextRelease',
+    'code_readiness': 'Release',
```
Comment (astarche) [inline source/codegen/metadata/nirfmxpulse/config.py:30]: Work with @ThangamV-NI to make sure that testing is sufficient for Release.

**Comment by astarche** [comment]
reckenro and maxxboehme are out so you do not need to wait on them. Please get review from @ThangamV-NI if possible.

**Review by astarche — APPROVED** [review]

File: source/codegen/metadata/nirfmxpulse/config.py (hunk 1 above)
Comment (ThangamV-NI) [reply source/codegen/metadata/nirfmxpulse/config.py:30]: Can you add session & driver-api tests, similar to other standards?

Comment (RahulBenaka) [reply source/codegen/metadata/nirfmxpulse/config.py:30]: I have added those test cases as part of the PR: #1111 by referring VNA personality
You can check the description for the successful test runs.

**Review by ThangamV-NI — APPROVED** [review]

**Pull request merged**